### PR TITLE
Allow concurrent use of el.cheapo

### PR DESCRIPTION
**Pull request by graetzer**
relevant port of https://github.com/arangodb/arangodb/pull/11854

http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/10689/

### CODE REVIEW

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/10689/

**Comment by graetzer** [comment]
http://172.16.10.101:8080/view/Java/job/arangodb-ANY-java-driver-matrix/942/

**Comment by rashtao** [comment]
> http://172.16.10.101:8080/view/Java/job/arangodb-ANY-java-driver-matrix/942/

http://172.16.10.101:8080/view/Java/job/arangodb-ANY-java-driver-matrix/943/

File: arangod/Transaction/ClusterUtils.cpp [hunk 1]
```diff
@@ -35,7 +35,23 @@ namespace arangodb {
 namespace transaction {
 namespace cluster {
 
+ void abortTransactions(LogicalCollection& coll) {
+   transaction::Manager* mgr = transaction::ManagerFeature::manager();
+   if (!mgr) { // might be called during shutdown
```
Comment (danielhlarkin) [inline arangod/Transaction/ClusterUtils.cpp:40]:
```suggestion
   if (!mgr) { // should only happen during unit tests
```

File: arangod/Transaction/Manager.cpp [hunk 2]
```diff
@@ -495,10 +495,22 @@ Result Manager::createManagedTrx(TRI_vocbase_t& vocbase, TRI_voc_tid_t tid,
...
+  TRI_ASSERT(mode != AccessMode::Type::NONE);
   auto& server = application_features::ApplicationServer::server();
   if (_disallowInserts.load(std::memory_order_acquire)) {
     return nullptr;
   }
+  
+  auto const role = ServerState::instance()->getRole();
+  std::chrono::steady_clock::time_point endTime;
+  if (!ServerState::isDBServer(role)) { // keep end time as small as possible
+    endTime = std::chrono::steady_clock::now() + std::chrono::seconds(8);
```
Comment (danielhlarkin) [inline arangod/Transaction/Manager.cpp:507]: Why 8 seconds? Should this be configurable via parameter or something?

**Comment by rashtao** [comment]
Java tests are successful:
http://172.16.10.101:8080/view/Java/job/arangodb-ANY-java-driver-matrix/952/

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/10752/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**